test/framework/batch_helper.hpp

```diff
@@ -195,6 +195,97 @@ namespace framework {
       return framework::expected::val(result_batch).value().value;
     }
 
+    /**
+     * Namespace provides useful functions which are related to implementation
+     * but they are internal API
+     */
+    namespace internal {
```

Comment:
We can split this file to .cpp and .hpp and then put all from internal into unnamed namespace in .cpp file or make all functions from `internal` static to .cpp file. Thus you can hide all methods inside

Reply:
My point is not to hide those methods but just mark them. Those methods are useful in different cases of creation batches.